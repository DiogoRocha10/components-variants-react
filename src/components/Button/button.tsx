import styles from "./button.module.css"
import cn from "clsx"

interface ButtonProps {
	variant?: "primary" | "secondary" | "inverse"
	backgroundColor?: string
	size?: "sm" | "md" | "lg"
	className?: string
	label: string
	onClick?: () => void
}

export const Button = ({
	variant = "primary",
	size = "md",
	backgroundColor,
	label,
	className,
	...props
}: ButtonProps) => {
	const sizeClasses = {
		[styles["-button--small"]]: size === "sm",
		[styles["-button--medium"]]: size === "md",
		[styles["-button--large"]]: size === "lg",
	}

	const variantClasses = {
		[styles["-button--primary"]]: variant === "primary",
		[styles["-button--secondary"]]: variant === "secondary",
		[styles["-button--inverse"]]: variant === "inverse",
	}

	return (
		<button
			type="button"
			className={cn(className, styles["-button"], variantClasses, sizeClasses)}
			style={{ backgroundColor }}
			{...props}
		>
			{label}
		</button>
	)
}
