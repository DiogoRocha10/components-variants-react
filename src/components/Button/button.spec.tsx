import { render } from "@testing-library/react"
import { Button } from "./button"
import renderer from "react-test-renderer"

it("text", () => {
	const { getByText } = render(<Button label="Button" />)
	expect(getByText("Button")).toBeTruthy()
})

it("renders correctly", () => {
	const tree = renderer.create(<Button label="Button" />).toJSON()
	expect(tree).toMatchSnapshot()
})
