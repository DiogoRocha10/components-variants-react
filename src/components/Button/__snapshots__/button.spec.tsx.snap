// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="-button -button--primary -button--medium"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
  type="button"
>
  Button
</button>
`;
